processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -415,23 +413,15 @@ public IndexedInts getRow()
         if (indices == null || indices.length == 0) {
           final int nullId = getEncodedValue(null, false);
           if (nullId > -1) {
-            if (nullId < maxId) {
-              row = new int[] {nullId};
-              rowSize = 1;
-            } else {
-              // Choose to use ArrayBasedIndexedInts later, instead of EmptyIndexedInts, for monomorphism
-              row = IntArrays.EMPTY_ARRAY;
-              rowSize = 0;
-            }
+            row = new int[] {nullId};
+            rowSize = 1;
           }
         }
 
         if (row == null && indices != null && indices.length > 0) {
           row = new int[indices.length];
           for (int id : indices) {
-            if (id < maxId) {
-              row[rowSize++] = id;
-            }
+            row[rowSize++] = id;
           }
         }
```

Comment:
Sorry, I misunderstood the problem you were talking about because of the line number thing. It should still be fine though, since row being null would lead to `return ArrayBasedIndexedInts.of(null, 0)`, which returns `ArrayBasedIndexedInts.EMPTY`.